packages/desktop-client/src/components/filters/SavedFilterMenuButton.tsx

```diff
@@ -0,0 +1,180 @@
+import { NameFilter } from './NameFilter';
+
+export type SavedFilter = {
+  conditions?: RuleConditionEntity[];
+  conditionsOp?: string;
+  id?: string;
+  name: string;
+  status?: string;
+};
+
+export function SavedFilterMenuButton({
+  filters,
+  conditionsOp,
+  filterId,
+  onClearFilters,
+  onReloadSavedFilter,
+  filtersList,
+}: {
+  filters: RuleConditionEntity[];
+  conditionsOp: string;
+  filterId: SavedFilter;
+  onClearFilters: () => void;
+  onReloadSavedFilter: (savedFilter: SavedFilter, value?: string) => void;
+  filtersList: RuleConditionEntity[];
+}) {
+  const [nameOpen, setNameOpen] = useState(false);
+  const [adding, setAdding] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [err, setErr] = useState(null);
+  const [menuItem, setMenuItem] = useState('');
+  const [name, setName] = useState(filterId.name);
+  const id = filterId.id;
+  let savedFilter: SavedFilter;
+
+  const onFilterMenuSelect = async (item: string) => {
+    setMenuItem(item);
+    switch (item) {
+      case 'rename-filter':
+        setErr(null);
+        setAdding(false);
+        setMenuOpen(false);
+        setNameOpen(true);
+        break;
+      case 'delete-filter':
+        setMenuOpen(false);
+        await send('filter-delete', id);
+        onClearFilters();
+        break;
+      case 'update-filter':
+        setErr(null);
+        setAdding(false);
+        setMenuOpen(false);
+        savedFilter = {
+          conditions: filters,
+          conditionsOp,
+          id: filterId.id,
+          name: filterId.name,
+          status: 'saved',
+        };
+        await sendCatch('filter-update', {
+          state: savedFilter,
+          filters: [...filtersList],
+        });
+        onReloadSavedFilter(savedFilter, 'update');
+        break;
+      case 'save-filter':
+        setErr(null);
+        setAdding(true);
+        setMenuOpen(false);
+        setNameOpen(true);
+        break;
+      case 'reload-filter':
+        setMenuOpen(false);
+        savedFilter = {
+          ...savedFilter,
+          status: 'saved',
+        };
+        onReloadSavedFilter(savedFilter, 'reload');
+        break;
+      case 'clear-filter':
+        setMenuOpen(false);
+        onClearFilters();
+        break;
+      default:
+    }
+  };
+
+  async function onAddUpdate() {
+    if (adding) {
+      //create new flow
+      savedFilter = {
+        conditions: filters,
+        conditionsOp,
+        name,
+        status: 'saved',
+      };
+      const res: string = await sendCatch('filter-create', {
+        state: savedFilter,
+        filters: [...filtersList],
+      });
+      savedFilter = {
+        ...savedFilter,
+        id: res,
+      };
+    } else {
+      //rename flow
+      savedFilter = {
+        conditions: filterId.conditions,
+        conditionsOp: filterId.conditionsOp,
+        id: filterId.id,
+        name,
+      };
+      await sendCatch('filter-update', {
```

Comment:
are we intentionally no longer caring about errors on these?

Reply:
No, not sure what happened there. I've added it back in and re-wrote it a bit.